File: CS/Scaffolding.ManyToMany/SchoolContextDataModel/SchoolContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.ManyToMany.Common.Utils;
using Scaffolding.ManyToMany.Common.DataModel;
using Scaffolding.ManyToMany.Common.DataModel.DesignTime;
using Scaffolding.ManyToMany.Common.DataModel.EntityFramework;
using Scaffolding.ManyToMany.Model;

namespace Scaffolding.ManyToMany.SchoolContextDataModel {
    /// <summary>
    /// A SchoolContextUnitOfWork instance that represents the run-time implementation of the ISchoolContextUnitOfWork interface.
    /// </summary>
    public class SchoolContextUnitOfWork : DbUnitOfWork<SchoolContext>, ISchoolContextUnitOfWork {

        public SchoolContextUnitOfWork(Func<SchoolContext> contextFactory)
            : base(contextFactory) {
        }

        IRepository<Course, int> ISchoolContextUnitOfWork.Courses {
            get { return GetRepository(x => x.Set<Course>(), (Course x) => x.CourseId); }
        }

        IRepository<Student, int> ISchoolContextUnitOfWork.Students {
            get { return GetRepository(x => x.Set<Student>(), (Student x) => x.StudentId); }
        }
    }
}

File: VB/Scaffolding.ManyToMany/SchoolContextDataModel/SchoolContextUnitOfWork.vb
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.ManyToMany.Common.Utils
Imports Scaffolding.ManyToMany.Common.DataModel
Imports Scaffolding.ManyToMany.Common.DataModel.EntityFramework
Imports Scaffolding.ManyToMany.Model

Namespace Scaffolding.ManyToMany.SchoolContextDataModel
    ''' <summary>
    ''' A SchoolContextUnitOfWork instance that represents the run-time implementation of the ISchoolContextUnitOfWork interface.
    ''' </summary>
    Public Class SchoolContextUnitOfWork
        Inherits DbUnitOfWork(Of SchoolContext)
        Implements ISchoolContextUnitOfWork

        Public Sub New(ByVal contextFactory As Func(Of SchoolContext))
            MyBase.New(contextFactory)
        End Sub

        Private ReadOnly Property ISchoolContextUnitOfWork_Courses() As IRepository(Of Course, Integer) Implements ISchoolContextUnitOfWork.Courses
            Get
                Return GetRepository(Function(x) x.Set(Of Course)(), Function(x) x.CourseId)
            End Get
        End Property

        Private ReadOnly Property ISchoolContextUnitOfWork_Students() As IRepository(Of Student, Integer) Implements ISchoolContextUnitOfWork.Students
            Get
                Return GetRepository(Function(x) x.Set(Of Student)(), Function(x) x.StudentId)
            End Get
        End Property
    End Class
End Namespace
